src/geo/transform.js

```diff
@@ -1737,6 +1742,9 @@ class Transform {
         if (!m) throw new Error("failed to invert matrix");
         this.pixelMatrixInverse = m;
 
+        // globe matrix
+        this.globeMatrix = this.projection.name === 'globe' ? calculateGlobeMatrix(this) : m;
```

Comment:
I think leaving it to `null`/`undefined` would be a good safeguard against accidental use of this matrix when not on on globe, and signal inconsistent use of it. 

But I'm not sure how graceful that would be with flow, as it would change the type to optional and we would have to wrap it up with checks.

```suggestion
        this.globeMatrix = this.projection.name === 'globe' ? calculateGlobeMatrix(this) : null;
```